### PR TITLE
creates book-an-accommodation controller.

### PR DESCRIPTION
**Pull request by minlinx**
#### What does this PR do?
* Creates a controller for users to book  an accommodation.
#### Description of Task to be completed?
* A Barefoot-Nomad user should be able to book an accommodation facility.
#### How should this be manually tested?
* Clone the repo and `cd` into the project directory
* Run `yarn install` to install the project's dependency
* Run `yarn start` to start the app
* On Postman, send a post request to http://localhost:3000/api/v1/accommodation.
#### What are the relevant pivotal tracker stories?
[#PT-Story-ID](http://PT-Story-Link)
https://www.pivotaltracker.com/n/projects/2381976/stories/167727670

### CODE REVIEW

File: src/migrations/02-create-accommodations.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  up: (queryInterface, Sequelize) => queryInterface.createTable('Accommodations', {
+  up: (queryInterface, Sequelize) => queryInterface.createTable('Rooms', {
```
Comment (chrismeeky) [inline src/migrations/02-create-accommodations.js:2]: Nice job. Why did you change all instances of Accommodation to Rooms?

File: src/routes/accommodationRoute.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { AccommodationController } from '../controllers';
+import { Authorization } from '../middlewares';
+import Validate from '../validation';
+
+const accommodationRoute = app => {
+  app.post(
+    '/api/v1/accommodation',
+    Validate.validateUserInput,
+    Authorization.checkToken,
```
Comment (chrismeeky) [inline src/routes/accommodationRoute.js:9]: you might consider adding the `confirmUser` middleware

File: src/migrations/02-create-accommodations.js (hunk 1 above)
Comment (chrismeeky) [reply src/migrations/02-create-accommodations.js:2]: Also consider modifying your PR title

Comment (chrismeeky) [reply src/migrations/02-create-accommodations.js:2]: Fix all hound errors

**Review by chrismeeky — APPROVED** [review]

File: src/migrations/02-create-accommodations.js (hunk 1 above)
Comment (minlinx) [reply src/migrations/02-create-accommodations.js:2]: I kept getting the **Relation does not exist** error till I did that.
I will checkout suitable titles. Can you suggest any?
I think, I will just resolve the **hound** issues here.

File: src/controllers/RequestController.js [hunk 3]
```diff
@@ -361,8 +361,9 @@ class RequestController {
       return HelperMethods.clientError(res,
         'The request you are trying to confirm cannot be found', 404);
     } catch (error) {
-      if (error.errors) return HelperMethods.sequelizeValidationError(res, error);
-      return HelperMethods.serverError(res);
+      // if (error.errors) return HelperMethods.sequelizeValidationError(res, error);
+      // return HelperMethods.serverError(res);
+      return next(error)
```
[the remark this answers is not in the document]
Comment (minlinx) [reply src/controllers/RequestController.js:366]: Can anybody explain why the test did not fail here?
I do nor know what I was looking for in this file!!!

File: src/controllers/AccommodationController.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import models from '../models';
+import { HelperMethods } from '../utils';
+
+const { Room } = models;
```
Comment (Daymorelah) [inline src/controllers/AccommodationController.js:4]: I can't remember creating a room model. what is going on here?!!

**Review by Daymorelah — CHANGES_REQUESTED** [review]
